tfjs-backend-webgl/src/kernels/Conv2D.ts

```diff
@@ -41,6 +42,18 @@ export function conv2d(
       convInfo.strideHeight === 1 && convInfo.strideWidth === 1 &&
       (convInfo.padInfo.type === 'SAME' || convInfo.padInfo.type === 'VALID')) {
     out = conv2dByMatMul({x, filter, convInfo, backend});
+  } else if (convInfo.strideWidth <= 2 && $dataFormat === 'channelsLast'
```

Comment:
Please pay attention that we may need more strict conditions to go to this path. For example, convInfo.dilationWidth === 1, convInfo.filterWidth > 3. For such kind of conv2d, `Conv2DPackedProgram` may get more benefits. Also try the impact of large/small convInfo.inChannels. We need better to know the threshold to go to Conv2DPackedProgram path in future.

Reply:
Thank you Jiajia! I am not clear why we need more strict conditions here?

As we known, Conv2DPackedProgram is more efficient than the following two branches (`conv2dWithIm2Row` and `Conv2DProgram`), because it minimizes the number of 'read' instructions (such as `getX`, `getW`) in the shaders.

Currently Conv2DPackedProgram only implements NHWC format. However, NCHW format works better for larger filters, so we may implement it in future. Do you mean add more strict conditions to let NCHW-Conv2DPackedProgram to work on these cases?

(Since we are discussing the future work, I merged this PR.)

Reply:
Personally, I don't think `Conv2DPackedProgram` is always better than `conv2dWithIm2Row` even you have the condition that `convInfo.strideWidth <= 2`. But currently, almost all the models in benchmarks satisfy this condition. So you can easily test this new algorithm in benchmarks. I tested your PR locally with resnet50 model in benchmarks on Intel CFL. It becomes worse when `WEBGL_EXP_CONV` is true. So I think the conditions should be changed, at least on intel CFL. I use resnet50 because it has many non-point wise conv2d, which can better see the performance difference. That's why I suspect whether we should have more strict conditions. Have you tested some benchmarks and see obviously advantages to use `Conv2DPackedProgram` compared with `conv2dWithIm2Row`?

Reply:
Thank you Jiajia for benchmarking this!

From my original perspective, `Conv2DPackedProgram ` aims to optimize all `non-point wise conv2d` against `conv2dWithIm2Row` (from your comments, this point of view deserves more experiments to support). The models (whose inference time is dominated by non-pointwise conv2d) are supposed to perform better if  set WEBGL_EXP_CONV as true.

`Conv2DPackedProgram` performs great for one of our internal models (whose inference time is dominated by non-pointwise conv2d) (the model is speed up from 170ms to 50ms). A comprehensive benchmarking is a great idea and I will do it and share with you later. 

I just tested bodypix with ResNet50 base on my Mac, and it just performance similar under `WEBGL_EXP_CONV` as true and as false. (The aggregated Conv2d time drops from 73.09ms to 52.11ms, when set it as true. It's wired that the final inference time keeps similar. I will take a look over this)
![image](https://user-images.githubusercontent.com/40653845/180588738-68eb5d85-08f4-46ed-a0e1-3c405f156834.png)
Since I do not have Intel CFL model, I am curious how worse it is for Intel CFL?

Reply:
@Linchenn Thanks for your data. I rechecked it on Intel CFL, TGL and NV RTX3090 on Windows. All of them have obvious regressions on `bodypix-ResNet50-image-0.5`.  Since `Conv2DPackedProgram` is behind a flag, it's ok now as it is. But once we plan to enable it by default. We need really to be careful about the conditions when to use it.

Take Intel TGL as an example:

- **WEBGL_EXP_CONV as true**

![webgl_exp_conv_true](https://user-images.githubusercontent.com/4221210/180692470-24d0cd05-9749-4584-9ac6-5371fa59c77f.png)
![webgl_exp_conv_true_avg_time](https://user-images.githubusercontent.com/4221210/180692505-0315bcab-4b62-4cc1-8b79-039ff877eb8d.png)
![webgl_exp_env_true_individual](https://user-images.githubusercontent.com/4221210/180692524-ccba4f70-a428-47f3-a218-22913be91446.png)

- WEBGL_EXP_CONV as false

![webgl_exp_conv_false](https://user-images.githubusercontent.com/4221210/180692553-ba60cc08-bc50-41e9-9020-90e222011831.png)
![webgl_exp_conv_false_avg_time](https://user-images.githubusercontent.com/4221210/180692572-6c98b706-732b-49c0-b009-865cf2f14efb.png)
![webgl_exp_conv_false_individual](https://user-images.githubusercontent.com/4221210/180692587-8c168d2a-8388-4dbf-8045-d94acabfcda8.png)

Reply:
Thank you Jiajia for the detailed checking on various devices! I will check it before removing the flag and share with yoiu latter.